torch/distributed/_tensor/device_mesh.py

```diff
@@ -162,11 +165,16 @@ def _get_or_create_default_group(self):
                 # automatically set the current cuda device base on num of gpu devices available in each host
                 # NOTE: This device selection would only work for homogeneous hardware.
                 torch.cuda.set_device(get_rank() % torch.cuda.device_count())
-            # TODO (xilunwu): to perform DTensor random ops, we need to ensure all ranks in mesh is initialized
-            # with the same random seed. The seed to use will be the current seed on rank 0. We store this seed
-            # as an attribute of device mesh for future use. However, the detail is still TBD how we gonna use
-            # this attribute, so we will implement this logic once we figure out the answer.
+            # To perform DTensor random ops, we need to ensure all ranks in mesh is initialized
+            # with the same random seed. The seed to use will be a random number sampled on rank 0.
+            # We store this seed as an attribute of device mesh for DTensor's random ops use only.
             self._seed = torch.cuda.initial_seed()
+            random.seed(self._seed)
+            obj_list = [random.randint(0, sys.maxsize)]
+            broadcast_object_list(obj_list)
+            # separate DTensor use only RNG state from the default RNG state
+            self._seed = int(obj_list[0])
+            self._offset = 0
```

Comment:
Also, I do think we should have sth like RNGStateTracker to abstract out all those logic out of device_mesh.py, directly embedding it in device_mesh.py is not a good practice as they are two different concepts.